### ISSUE TITLE
[Bug]: Apply changes popup infinite loop

### ISSUE THREAD

**Issue by dt192** [sqlitebrowser/sqlitebrowser#3758]
### What did you do?

Try clearing a filter without applying a change, then clicking the apply button on the popup.

### What did you expect to see?

The popup to go

### What did you see instead?

It keeps reappearing every time I press apply and if I hold down the enter key the app crashes.

### DB4S Version

3.13.99 (nightly)

### What OS are you seeing the problem on?

Windows

### OS version

Win 10

### Prevention against duplicate issues

- [X] I have searched for similar issues

**Comment by justinclift** [comment]
That definitely sounds like a bug. :frowning:

Thanks for reporting it @dt192. :smile:

**Comment by mgrojo** [comment]
I was unable to reproduce, please provide more details.

**Comment by dt192** [comment]
https://github.com/user-attachments/assets/b3694aaa-e157-42aa-b0f4-5992551d5525

**Comment by mgrojo** [comment]
The only minor problem that I reproduce is that removing the filter after the prompt dialog requires a second click. It might be Windows specific.

* Does it happen on any DB and any column?
* Is there some relevant error in the "Error Log" ("SQL Log" dock)

**Comment by dt192** [comment]
Yes, same with a simple example. Are you editing it in the Edit Database Cell panel? It doesn't happen if you edit in place.

https://github.com/user-attachments/assets/480209f1-8b18-480a-9678-3e9d4f190a8f

It also happens like this

![2024-10-14_23-07](https://github.com/user-attachments/assets/23a82271-a293-48f0-8bd4-15a9cbb91456)

**Comment by dt192** [comment]
@mgrojo That solves the loop issue, it does apply/discard correctly. The issue now is that the filter does not get removed.

**Comment by mgrojo** [comment]
> @mgrojo That solves the loop issue, it does apply/discard correctly. The issue now is that the filter does not get removed.

Great, then the behaviour is already the [same as under Linux](https://github.com/sqlitebrowser/sqlitebrowser/issues/3758#issuecomment-2412364306). That's less critical, it can wait.
